> 식당데이터 : justmarkham/DAT8

> DataUrl = ‘https://raw.githubusercontent.com/Datamanim/pandas/main/chipo.csv’

Question 20

데이터를 로드하라.

In [2]:
import pandas as pd

dataPath = '../data/chipo.csv'

df = pd.read_csv(dataPath)

# df.to_csv('../data/chipo.csv')

df

order_id  quantity                              item_name   
0            1         1           Chips and Fresh Tomato Salsa  \
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

Question 21

quantity컬럼 값이 3인 데이터를 추출하여 첫 5행을 출력하라

In [ ]:
isThree = df['quantity'] == 3

three = df[isThree]

three.head()

# df.loc[df['quantity'] == 3]



Question 22

quantity컬럼 값이 3인 데이터를 추출하여 index를 0부터 정렬하고 첫 5행을 출력하라

In [ ]:
isThree = df['quantity'] == 3

three = df[isThree]

three.reset_index(inplace=True)

three

Question 23

quantity , item_price 두개의 컬럼으로 구성된 새로운 데이터 프레임을 정의하라

In [ ]:
newdf = df[['quantity', 'item_price']]

newdf

Question 24

item_price 컬럼의 달러표시 문자를 제거하고 float 타입으로 저장하여 new_price 컬럼에 저장하라

In [ ]:
df['new_price'] = df['item_price'].str.replace('$', '')

# dateFrame astype은 inplace가 안되고 변수에 다시 대입해줘야 한다.
df = df.astype({'new_price' : 'float'})

df.dtypes

Question 25

new_price 컬럼이 5이하의 값을 가지는 데이터프레임을 추출하고, 전체 갯수를 구하여라

In [ ]:
isFive = df['new_price'] <= 5

fiveValue = df[isFive]

len(fiveValue)

Question 26

item_name명이 Chicken Salad Bowl 인 데이터 프레임을 추출하라고 index 값을 초기화 하여라

In [ ]:
isChickenBowl = df['item_name'] == 'Chicken Salad Bowl'

chickenBowl = df[isChickenBowl]

chickenBowl.reset_index()

Question 27

new_price값이 9 이하이고 item_name 값이 Chicken Salad Bowl 인 데이터 프레임을 추출하라

In [ ]:
# and 조건으로 비교하는 방법을 생각해 보자
# isChickenBowl = df['item_name'] == 'Chicken Salad Bowl'

# chickenBowl = df[isChickenBowl]

# isChickenBowl9 = chickenBowl['new_price'] <= 9

# chickenBowl9 = chickenBowl[isChickenBowl9]

# chickenBowl9

# 여러 컬럼의 값을 and, or 로 비교하려면 ()안에 조건식이 들어가야 된다
# () 사용 안하면 에러 발생
isChickenBowl = (df['item_name'] == 'Chicken Salad Bowl') & (df['new_price'] <= 9)

chickenBowl = df[isChickenBowl]

chickenBowl

Question 28

df의 new_price 컬럼 값에 따라 오름차순으로 정리하고 index를 초기화 하여라

In [ ]:
# sort_Values(컬럼명, ascending = False) 
#ascending 이 False면 내림차순. 기본값은 오름차순
df.sort_values('new_price')

Question 29

df의 item_name 컬럼 값중 Chips 포함하는 경우의 데이터를 출력하라

In [ ]:
dataFilter = df['item_name'].str.contains('Chips')

dataChips = df[dataFilter]

dataChips

Question 30

df의 짝수번째 컬럼만을 포함하는 데이터프레임을 출력하라

In [ ]:
df.iloc[ : , ::2]

Question 31

df의 new_price 컬럼 값에 따라 내림차순으로 정리하고 index를 초기화 하여라

In [ ]:
df.sort_values('new_price', ascending=False).reset_index()

Question 32

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 인덱싱하라

In [ ]:
df[(df['item_name'] == 'Steak Salad') | (df['item_name'] == 'Bowl')]

Question 33

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 데이터 프레임화 한 후, item_name를 기준으로 중복행이 있으면 제거하되 첫번째 케이스만 남겨라

In [ ]:
# DataFrame 에서 비교 연산자는 and, or, not으로 사용 하면 안되고 &, |, ~ 이렇게 사용 해야 된다.

isnewdf = (df['item_name'] == 'Steak Salad') | (df['item_name'] == 'Bowl')

newdf = df[isnewdf]

# newdf['item_name'].unique()

newdf.drop_duplicates('item_name')

Question 34

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 데이터 프레임화 한 후, item_name를 기준으로 중복행이 있으면 제거하되 마지막 케이스만 남겨라

In [ ]:
isnewdf = (df['item_name'] == 'Steak Salad') | (df['item_name'] == 'Bowl')

newdf = df[isnewdf]

newdf.drop_duplicates('item_name', keep = 'last')

Question 35

df의 데이터 중 new_price값이 new_price값의 평균값 이상을 가지는 데이터들을 인덱싱하라

In [ ]:
df.loc[df.new_price >= df.new_price.mean()]

Question 36

df의 데이터 중 item_name의 값이 Izze 데이터를 Fizzy Lizzy로 수정하라

In [5]:
df.loc[df.item_name == 'Izze', 'item_name'] = 'Fizzy Lizzy'

df

order_id  quantity                              item_name   
0            1         1           Chips and Fresh Tomato Salsa  \
1            1         1                            Fizzy Lizzy   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                NoData     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                NoData     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

Question 37

df의 데이터 중 choice_description 값이 NaN 인 데이터의 갯수를 구하여라

In [9]:
df.choice_description.isnull().sum()

0

Question 38

df의 데이터 중 choice_description 값이 NaN 인 데이터를 NoData 값으로 대체하라(loc 이용)

In [3]:
df.loc[df.choice_description.isnull(), 'choice_description'] = 'NoData'

df



order_id  quantity                              item_name   
0            1         1           Chips and Fresh Tomato Salsa  \
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                NoData     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                NoData     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

Question 39

df의 데이터 중 choice_description 값에 Black이 들어가는 경우를 인덱싱하라

In [6]:
df.loc[df['choice_description'].str.contains('Black')]

order_id  quantity             item_name   
4            2         2          Chicken Bowl  \
7            4         1         Steak Burrito   
9            5         1         Steak Burrito   
11           6         1  Chicken Crispy Tacos   
12           6         1    Chicken Soft Tacos   
...        ...       ...                   ...   
4604      1828         1          Chicken Bowl   
4608      1829         1        Veggie Burrito   
4611      1830         1        Veggie Burrito   
4612      1831         1         Carnitas Bowl   
4617      1833         1         Steak Burrito   

                                     choice_description item_price  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
7     [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
9     [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25   
11    [Roasted Chili Corn Salsa, [Fajita Vegetables,...     $8.75   
12    [Roasted Chili Corn Salsa, [Rice, Black Beans,...     $8.75   
...                                                 ...        ...  
4604  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...     $8.75   
4608  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.25   
4611  [Tomatillo Green Chili Salsa, [Rice, Fajita Ve...    $11.25   
4612  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...     $9.25   
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   

[1353 rows x 5 columns]

Question 40

df의 데이터 중 choice_description 값에 Vegetables 들어가지 않는 경우의 갯수를 출력하라

In [10]:
len(df.loc[~(df['choice_description'].str.contains('Vegetables'))])

3900

Question 41

df의 데이터 중 item_name 값이 N으로 시작하는 데이터를 모두 추출하라

In [11]:
#startswith()
df.loc[df['item_name'].str.startswith('N')]

order_id  quantity         item_name         choice_description   
2            1         1  Nantucket Nectar                    [Apple]  \
22          11         1  Nantucket Nectar       [Pomegranate Cherry]   
105         46         1  Nantucket Nectar  [Pineapple Orange Banana]   
173         77         1  Nantucket Nectar                    [Apple]   
205         91         1  Nantucket Nectar             [Peach Orange]   
436        189         1  Nantucket Nectar       [Pomegranate Cherry]   
601        247         2  Nantucket Nectar  [Pineapple Orange Banana]   
925        381         1  Nantucket Nectar       [Pomegranate Cherry]   
1356       553         1  Nantucket Nectar       [Pomegranate Cherry]   
1585       641         1  Nantucket Nectar             [Peach Orange]   
1626       656         1  Nantucket Nectar  [Pineapple Orange Banana]   
1706       690         1  Nantucket Nectar                    [Apple]   
2162       872         1  Nantucket Nectar  [Pineapple Orange Banana]   
2379       947         2  Nantucket Nectar             [Peach Orange]   
2381       947         1  Nantucket Nectar                    [Apple]   
2430       965         1  Nantucket Nectar       [Pomegranate Cherry]   
2653      1053         1  Nantucket Nectar  [Pineapple Orange Banana]   
2818      1118         1  Nantucket Nectar                    [Apple]   
2838      1128         1  Nantucket Nectar             [Peach Orange]   
2853      1133         1  Nantucket Nectar                    [Apple]   
2949      1172         1  Nantucket Nectar             [Peach Orange]   
3318      1330         1  Nantucket Nectar             [Peach Orange]   
3368      1351         1  Nantucket Nectar  [Pineapple Orange Banana]   
3570      1433         1  Nantucket Nectar  [Pineapple Orange Banana]   
3845      1541         1  Nantucket Nectar             [Peach Orange]   
4019      1609         1  Nantucket Nectar  [Pineapple Orange Banana]   
4078      1632         1  Nantucket Nectar             [Peach Orange]   

     item_price  
2        $3.39   
22       $3.39   
105      $3.39   
173      $3.39   
205      $3.39   
436      $3.39   
601      $6.78   
925      $3.39   
1356     $3.39   
1585     $3.39   
1626     $3.39   
1706     $3.39   
2162     $3.39   
2379     $6.78   
2381     $3.39   
2430     $3.39   
2653     $3.39   
2818     $3.39   
2838     $3.39   
2853     $3.39   
2949     $3.39   
3318     $3.39   
3368     $3.39   
3570     $3.39   
3845     $3.39   
4019     $3.39   
4078     $3.39

Question 42

df의 데이터 중 item_name 값의 단어갯수가 15개 이상인 데이터를 인덱싱하라

In [ ]:
df.loc[df['item_name'].str.len() >= 15]

Question 43

df의 데이터 중 new_price값이 lst에 해당하는 경우의 데이터 프레임을 구하고 그 갯수를 출력하라 lst =[1.69, 2.39, 3.39, 4.45, 9.25, 10.98, 11.75, 16.98]